### DISCUSSION TITLE
citations in margins + citations within captions + LaTeX == trouble

### DISCUSSION THREAD

**Discussion by bbolker** [quarto-dev/quarto-cli#6686] [Q&A]
### Description

As far as I can tell this is actually a LaTeX problem of nesting a `\marginpar{}` inside a `\caption{}`, as per [this Stack Exchange question](https://tex.stackexchange.com/questions/211784/marginpar-inside-figure-environment-fails).  I was able to hack around it by changing `\marginpar{}` to `\marginnote{}` in the LaTeX file with the following R-script hack (then building PDF from LaTeX myself); I also needed to make sure the figure had a short caption (`figure-scap`) in addition to the one with a caption in it.

This is arguably not Quarto's fault, but it might bite someone else, and *might* be solvable by substituting `marginnote` for `marginpar` in appropriate places?

(I don't know if this should be an issue rather than a discussion.)

## R hack code

This could probably be done in one line of Perl, or a multi-line `sed` substitution, or ...

```
cc <- commandArgs(trailingOnly = TRUE)
search_range <- 4
fn <- cc[1]
invisible(file.copy(fn, paste0(fn, ".bak"), overwrite = TRUE))
r <- readLines(fn)
caption_locs <- grep("\\caption", r)
if (length(caption_locs) > 0 ) {
    for (L in caption_locs) {
        rng <- L:(L+search_range)
        r[rng] <- gsub("marginpar", "marginnote", r[rng])
    }
}
writeLines(r, fn)
```

## reproducible example

````
---
title: "test"
citation-location: margin
bibliography: "test.bib"
---

<!-- "From @boxScience1976b" -->
```{r boxfig, echo = FALSE}
#| fig-cap: no ref
plot(1:10, 1:10)
```
````

with a bib file `test.bib`:

Running `quarto render test.qmd --to latex; pdflatex test` gives

```
Runaway argument?
{From Box (1976)\marginpar {\begin {footnotesize}\leavevmode \vadjust \ETC.
! Paragraph ended before \caption@prepareanchor was complete.
<to be read again> 
                   \par 
l.192 ...949}.\vspace{2mm}\par\end{footnotesize}}}
```

The offending LaTeX code is 

```
\caption{From Box
(1976)\marginpar{\begin{footnotesize}\leavevmode\vadjust pre{\protect\hypertarget{ref-boxScience1976b}{}}%
Box, George E. P. 1976. {``Science and {Statistics}.''} \emph{Journal of
the American Statistical Association} 71 (356): 791--99.
\url{https://doi.org/10.1080/01621459.1976.10480949}.\vspace{2mm}\par\end{footnotesize}}}
```

* the problem goes away if I (1) drop `citation-location: margin` **or** (2) leave @boxScience1976b out of the caption **or** (3) as above, change `marginpar` to `marginnote` in the LaTeX file *and* make sure that a short caption is specified

This is on PopOS! (ubuntu derivative) 22.04, with Quarto 1.2.335 and pandoc 3.1.7.

**Comment by mcanouil** [comment]
Closing in favour of:
- #6716
